src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -223,13 +209,9 @@ public class XmlConvert
 
             if (first)
             {
-                if ((!XmlCharType.IsStartNCNameCharXml4e(name[0]) && (local || (!local && name[0] != ':'))) ||
-                     matchPos == 0)
+                if (!XmlCharType.IsStartNCNameCharXml4e(name[0]) && (local || name[0] != ':') || matchPos == 0)
```

Comment:
Please leave the extra brackets here - It's not as obvious what order those are executed